core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.blobstore;
+
+import com.google.common.collect.Maps;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.blobstore.BlobContainer;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.*;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Base class that handles serialization of various data structures during snapshot/restore operations.
+ */
+public abstract class BlobStoreFormat<T extends ToXContent> {
+
+    protected final String blobNameFormat;
+
+    protected final FromXContentBuilder<T> reader;
+
+    protected final ParseFieldMatcher parseFieldMatcher;
+
+    // Serialization parameters to specify correct context for metadata serialization
+    protected static final ToXContent.Params SNAPSHOT_ONLY_FORMAT_PARAMS;
+
+    static {
+        Map<String, String> snpashotOnlyParams = Maps.newHashMap();
+        snpashotOnlyParams.put(MetaData.CONTEXT_MODE_PARAM, MetaData.CONTEXT_MODE_SNAPSHOT);
```

Comment:
It's unclear what exactly is going on here, can you add a comment as to why this is needed and what it does?